src/main/msp/msp.c

```diff
@@ -1519,6 +1521,8 @@ case MSP_NAME:
         sbufWriteU16(dst, gpsSol.groundCourse);
         // Added in API version 1.44
         sbufWriteU16(dst, gpsSol.dop.pdop);
+        // Added in API version 1.47
+        sbufWriteU8(dst, gpsData.state);
```

Comment:
State is internal detail for GPS implementation, and quite poorly designed one. Currently, it does not reflect well actual GPS condition(for example,  RECEIVING_DATA will timeout periodically until GPS has fix) . Making it public will only complicate GPS code refactoring. This is fine for testing, but something more stable should be merged.
Better list states that user in interested in and map GPS internals to it.

Reply:
@ledvinap Probably I have misinterpreted

> Configurator should display gpsState...

https://github.com/betaflight/betaflight/pull/13852#issuecomment-2321440948


@ctzsnooze yes this is intended to measure connection state in configurator.
As configurator waits for gps sensor to be set to display the icon:
https://github.com/betaflight/betaflight/blob/master/src/main/io/gps.c#L1488

Reply:
@haslinghuis: Not misinterpreted. But it's more complicated than I thought. It's hard to make changes in GPS code without making the situation worse.

I'll look into extra states that will beter match GPS state (receiving valid packets/baudrate correct; receiving necessary type of packets/user configured GPS correctly; got fix; got fix matching HOME criteria)

Reply:
Thanks @ledvinap.
With MSP we need to know:
- that the MSP source thinks that it has communications with whatever device provides it with GPS data - ie that it has a functional hardware connection to its data source device
- whether or not it has a 3D fix
I think we already get sat count over MSP.
We don't really care about how the MSP source talks to its module, but we do care about the update interval (data packet interval), since I recall a number of MSP GPS providers who struggled to get 10Hz working, and some only send 1Hz.
We don't configure MSP devices, we just get their position data; it's up to the manufacturer of the MSP GPS device to provide us with the position data only, how they get it is entirely up to them.